tests/unit-tests/test-class-sensei-analysis-overview-list-table.php

```diff
@@ -280,4 +287,51 @@ public function testGetLastActivityDateByCourseLessons() {
 			'The last activity date format or timezone is invalid.'
 		);
 	}
+	/**
+	 * Tests that we are getting right data for Completion Rate column.
+	 *
+	 * @covers Sensei_Analysis_Overview_List_Table::get_row_data
+	 * @dataProvider dataFortestCompletionRateForLesson
+	 */
+	public function testCompletionRateForLesson( $enrolled_student_count, $completed_student_count, $expected_output ) {
+		$user_ids                    = $this->factory->user->create_many( $enrolled_student_count );
+		$course_lessons              = $this->factory->get_course_with_lessons(
+			array(
+				'lesson_count' => 1,
+			)
+		);
+		$instance                    = new Sensei_Analysis_Overview_List_Table();
+		$instance->type              = 'lessons';
+		$get_lessons_method          = new ReflectionMethod( $instance, 'get_lessons' );
+		$get_row_data_lessons_method = new ReflectionMethod( $instance, 'get_row_data' );
+		$get_lessons_method->setAccessible( true );
+		$get_row_data_lessons_method->setAccessible( true );
+		$lessons_args = array(
+			'offset'  => 0,
+			'number'  => -1,
+			'orderby' => '',
+			'order'   => 'ASC',
+		);
+		$lesson_id    = array_pop( $course_lessons['lesson_ids'] );
+		foreach ( $user_ids as $key => $user_id ) {
+			Sensei_Utils::sensei_start_lesson( $lesson_id, $user_id, $key < $completed_student_count );
+		}
+		$lessons  = $get_lessons_method->invoke( $instance, $lessons_args );
+		$row_data = $get_row_data_lessons_method->invoke( $instance, array_pop( $lessons ) );
+		$this->assertEquals( $row_data['completion_rate'], $expected_output );
```

Comment:
You could add an assert message to better describe what failed. Right now the message is `Sensei_Analysis_Overview_List_Table_Test::testCompletionRateForLesson with data set "100%" (5, 5, '101%')` which is not very descriptive.